bin/p2-label/main.go

```diff
@@ -0,0 +1,143 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/square/p2/pkg/kp"
+	"github.com/square/p2/pkg/kp/flags"
+	"github.com/square/p2/pkg/labels"
+	"github.com/square/p2/pkg/util"
+	"gopkg.in/alecthomas/kingpin.v2"
+	klabels "k8s.io/kubernetes/pkg/labels"
+)
+
+var (
+	cmdApply               = kingpin.Command(CmdApply, "Apply label changes to all objects matching a selector")
+	applyLabelType         = cmdApply.Flag("labelType", "The type of label to adjust. Sometimes called the \"label tree\"").Short('t').Required().String()
+	applySubjectSelector   = cmdApply.Flag("selector", "The selector on which to modify labels.").Short('s').Required().String()
+	applyAddititiveLabels  = cmdApply.Flag("add", "The label set to apply to the subject.").Short('a').String()
+	applyDestructiveLabels = cmdApply.Flag("delete", "The label set to remove to the subject. It is not an error to include extra labels here.").Short('d').String()
```

Comment:
I think that "deletes are idempotent" describes the concept more accurately. For instance, if I use --delete for a flag that doesn't apply to the some members of the subjectSelector set this program will have a 0 exit code (ie. it's not an error to retry this or include extra labels to remove).